src/arrayMethodJoin.js

```diff
@@ -4,8 +4,19 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const strSeparator = String(separator);
+    let result = '';
+
+    for (const str of this) {
+      if (str === undefined || str === null) {
+        result += strSeparator;
+      } else {
+        result += str + strSeparator;
+      }
```

Comment:
In both cases you'll add separator, let's move this action outside of if/else block